homeassistant/components/tuya/climate.py

```diff
@@ -89,21 +108,59 @@ def __init__(self, tuya, platform):
         super().__init__(tuya, platform)
         self.entity_id = ENTITY_ID_FORMAT.format(tuya.object_id())
         self.operations = [HVAC_MODE_OFF]
+        self._has_operation = False
+        self._def_hvac_mode = HVAC_MODE_AUTO
+        self._min_temp = None
+        self._max_temp = None
+        self._temp_entity = None
+        self._temp_entity_error = False
+
+    @callback
+    def _load_config(self):
+        """Set device config parameter."""
+        config = self._get_device_config()
+        if not config:
+            return
+        unit = config.get(CONF_UNIT_OF_MEASUREMENT)
+        if unit:
+            self._tuya.set_unit("FAHRENHEIT" if unit == TEMP_FAHRENHEIT else "CELSIUS")
+        self._tuya.temp_divider = config.get(CONF_TEMP_DIVIDER, 0)
+        self._tuya.curr_temp_divider = config.get(CONF_CURR_TEMP_DIVIDER, 0)
+        min_temp = config.get(CONF_MIN_TEMP, 0)
+        max_temp = config.get(CONF_MAX_TEMP, 0)
+        if min_temp >= max_temp:
+            self._min_temp = self._max_temp = None
+        else:
+            self._min_temp = min_temp
+            self._max_temp = max_temp
+        self._temp_entity = config.get(CONF_EXT_TEMP_SENSOR)
 
     async def async_added_to_hass(self):
         """Create operation list when add to hass."""
         await super().async_added_to_hass()
+        self._load_config()
+        self.async_on_remove(
+            async_dispatcher_connect(self.hass, SIGNAL_CONFIG_ENTITY, self._load_config)
+        )
+
         modes = self._tuya.operation_list()
         if modes is None:
+            if self._def_hvac_mode not in self.operations:
+                self.operations.append(self._def_hvac_mode)
             return
 
         for mode in modes:
             if mode in TUYA_STATE_TO_HA:
```

Comment:
```suggestion
            if mode not in TUYA_STATE_TO_HA:
                continue
```

Then outdent the block below